### PR TITLE
Parameter checksum

### PR DESCRIPTION
**Pull request by cristianfchq**
Add parameter checksum and save on database

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 1]
```diff
@@ -43,28 +45,26 @@
...
-        List<String> resultPath = projects.stream().filter(project -> project.getChecksum().equals(checksum))
-                .map(project -> project.getPath())
-                .collect(Collectors.toList());
+        final int userID = 1;
+        List<Asset> assets = listAsset();
+        List<String> resultTitle = getTitles(checksum, assets);
+        List<String> resultPath = getPath(checksum, assets);
+
         exist = resultTitle.size() > 0;
+
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: Delete Blank line

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 2]
```diff
@@ -43,28 +45,26 @@
...
-                .map(project -> project.getTitle())
-                .collect(Collectors.toList());
-        List<String> resultPath = projects.stream().filter(project -> project.getChecksum().equals(checksum))
-                .map(project -> project.getPath())
-                .collect(Collectors.toList());
+        final int userID = 1;
+        List<Asset> assets = listAsset();
+        List<String> resultTitle = getTitles(checksum, assets);
+        List<String> resultPath = getPath(checksum, assets);
+
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: Delete Blank line

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 3]
```diff
@@ -26,8 +27,9 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
-import static org.fundacion.jala.converter.models.ProjectSQL.insertProjectData;
-import static org.fundacion.jala.converter.models.ProjectSQL.listProject;
+
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: Delete Blank line

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 4]
```diff
@@ -84,22 +84,36 @@ public String uploadFile(@RequestParam("file") MultipartFile file,
...
-            Thread.sleep(WAIT_TIME);
+            Thread.sleep(waitTime);
             zipFile(pathFile + outputFilename, pathFile + nameWithoutExtension + "zip");
+            System.out.println();
         }
         final String baseUrl = ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString();
         String downloadLink = baseUrl + "/api/download/" + nameWithoutExtension + "zip";
         return downloadLink;
     }
+
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java:107]: Javadoc is missing.

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 5]
```diff
@@ -84,22 +84,36 @@ public String uploadFile(@RequestParam("file") MultipartFile file,
...
+    private List<String> getPath(final String checksum, final List<Asset> assets) {
+        return assets.stream().filter(project -> project.getChecksum().equals(checksum))
+                    .map(asset -> asset.getPath())
+                    .collect(Collectors.toList());
+    }
+
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java:119]: Javadoc is missing.

**Review by JorgeRCV — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/view/converter/AudioConverterInterface.java [hunk 6]
```diff
@@ -119,26 +121,29 @@ public void actionPerformed(final ActionEvent e)  {
...
-        AudioRequestForm audioRequestForm = new AudioRequestForm();
-        audioRequestForm.addFilepath(file.getOriginFilePath());
-        audioRequestForm.addFormat(audioSelect.getConvertTo());
-        audioRequestForm.addBitrate(bitrate);
-        audioRequestForm.addVolume(settings.getVolume());
-        audioRequestForm.addHz(settings.getHz());
-        audioRequestForm.addAudiochannel(settings.getAudioChannel());
-        audioRequestForm.addMetadata(String.valueOf(settings.isMetadata()));
-        clientRequest.executeRequest(audioRequestForm);
+
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/converter/AudioConverterInterface.java]: Add Blank line between 117 and 118

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 7]
```diff
@@ -26,8 +27,9 @@
...
 import java.util.List;
 import java.util.stream.Collectors;
-import static org.fundacion.jala.converter.models.ProjectSQL.insertProjectData;
-import static org.fundacion.jala.converter.models.ProjectSQL.listProject;
+
+import static org.fundacion.jala.converter.models.AssetSQL.insertAssetData;
+import static org.fundacion.jala.converter.models.AssetSQL.listAsset;
 import static org.fundacion.jala.converter.service.ChecksumService.getFileChecksum;
 import static org.fundacion.jala.converter.service.ExtractMetadata.extractMetadata;
 import static org.fundacion.jala.converter.service.ZipService.*;
```
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: Specify the service, don't use "*"

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 8]
```diff
@@ -84,22 +84,36 @@ public String uploadFile(@RequestParam("file") MultipartFile file,
...
         String nameWithoutExtension = outputFilename.substring(0, outputFilename.lastIndexOf(".") + 1);
         extractMetadata(metadata, outputFilename, fileStorageService);
-        String pathFile = storagePath.substring(0, storagePath.lastIndexOf("\\") + 1);
+        String pathFile = storagePath.substring(0, storagePath.lastIndexOf(System.getProperty("file.separator")) + 1);
+
         if (!(resultTitle.size() > 0)) {
-            insertProjectData(filename, pathFile, checksumLocal, USER_ID);
+            insertAssetData(filename, pathFile, checksumLocal, userID);
         }
         if (metadata.equals("true")) {
```
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java:92]: Maybe this is redundant?, metadata should be boolean.

File: src/main/java/org/fundacion/jala/converter/view/converter/AudioConverterInterface.java [hunk 9]
```diff
@@ -5,6 +5,8 @@
  * ("Confidential Information"). You shall not disclose such Confidential
  * Information and shall use it only in accordance with the terms of the
  * license agreement you entered into with Fundacion Jala
+ *
```
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/view/converter/AudioConverterInterface.java:8]: missing author

**Review by Paola-Xaq — APPROVED** [review]
There are some details missing, but everything seems ok.

**Review by GustavoZHA — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java (hunk 3 above)
Comment (cristianfchq) [reply src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: ok

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java (hunk 7 above)
Comment (cristianfchq) [reply src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: ok, I will resolve that

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java (hunk 2 above)
Comment (cristianfchq) [reply src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: ok

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java (hunk 1 above)
Comment (cristianfchq) [reply src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: ok

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java (hunk 8 above)
Comment (cristianfchq) [reply src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java:92]: yes, but the metadata variable is of type String in this class when the parameter is received

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java (hunk 4 above)
Comment (cristianfchq) [reply src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java:107]: ok

**Pull request merged**